casper/src/test/scala/coop/rchain/casper/api/LastFinalizedAPITest.scala

```diff
@@ -0,0 +1,145 @@
+package coop.rchain.casper.api
+
+import cats.effect.Sync
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.casper.engine.Engine
+import coop.rchain.casper.{EngineWithCasper, SafetyOracle}
+import coop.rchain.casper.helper._
+import coop.rchain.casper.helper.BlockGenerator._
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.GenesisBuilder._
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.util.ConstructDeploy.basicDeployData
+import coop.rchain.crypto.signatures.Secp256k1
+import coop.rchain.metrics.Metrics
+import coop.rchain.shared.{Cell, Log}
+import coop.rchain.shared.scalatestcontrib._
+import monix.eval.Task
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+import monix.execution.Scheduler.Implicits.global
+
+import scala.collection.immutable.Map
+
+class LastFinalizedAPITest
+    extends FlatSpec
+    with Matchers
+    with EitherValues
+    with BlockGenerator
+    with BlockDagStorageFixture {
+  val validatorKeyPairs = (1 to 3).map(_ => Secp256k1.newKeyPair)
+  val (_, validatorPks) = validatorKeyPairs.unzip
+  val validators        = validatorPks.map(pk => ByteString.copyFrom(pk.bytes))
+  val v1                = validators(0)
+  val v2                = validators(1)
+  val v3                = validators(2)
+  val genesisParameters = buildGenesisParameters(
+    validatorKeyPairs,
+    Map(validatorPks(0) -> 10, validatorPks(1) -> 10, validatorPks(2) -> 5)
+  )
+  val genesisContext = buildGenesis(genesisParameters)
+
+  implicit val metricsEff = new Metrics.MetricsNOP[Task]
+
+  def isFinalized(block: BlockMessage)(engineCell: Cell[Task, Engine[Task]])(
+      implicit blockStore: BlockStore[Task],
+      safetyOracle: SafetyOracle[Task],
+      log: Log[Task]
+  ): Task[Boolean] =
+    BlockAPI
+      .isFinalized[Task](
+        IsFinalizedQuery(hash = ProtoUtil.hashString(block))
+      )(Sync[Task], engineCell, safetyOracle, blockStore, log)
+      .map(
+        _.right.value.isFinalized
+      )
+
+  /*
+   * DAG Looks like this:
+   *
+   *           b7
+   *           |
+   *           b6
+   *           |
+   *           b5 <- last finalized block
+   *         / |
+   *        |  b4
+   *        |  |
+   *       b2  b3
+   *         \ |
+   *           b1
+   *           |
+   *         genesis
+   */
+  "isFinalized" should "return true for ancestors of last finalized block" in effectTest {
+    HashSetCasperTestNode.networkEff(genesisContext, networkSize = 3).use {
+      case n1 +: n2 +: n3 +: _ =>
```

Comment:
this can be

`case nodes @ n1 +: n2 +: n3 +: Seq()`

and then we don't need to declare `val nodes  = List(n1, n2, n3)` below